condensation-forum/application.py

```diff
@@ -301,6 +314,117 @@ def logoutCallback():
     print("User signed out: " + user["name"], file=sys.stderr)
 
 
+@application.route('/file-delete', methods=['POST'])
+@authManager.requireAuthentication
+def fileManagerDeleteHander():
+    user = authManager.getUserData()
+    fid = int(request.form['file'])
+    
+    if not user:
+        abort(403)
+    id = user['id']
+
+    #delete the file by Cloud_key in AWS S3
+    with dataSessionMgr.session_scope() as dbSession:
+        file1 = query.getFileById(dbSession,fid)
+        file1 = query.extractOutput(file1)
+    
+    key = file1['cloud_key']
+    try:
+        s3client.delete_object(Bucket=bucket_name,Key=key)
+    except Exception as e:
+        print("Something happen: ",e)
+        return e
+    
+    #delete the file by fileID in DB
+    with dataSessionMgr.session_scope() as dbSession:
+        file = query.getFileById(dbSession,fid)
+        if file:
+            dbSession.delete(file)
+            flash("File Deleted")
+  
+    return redirect(url_for("fileManager"))
+
+@application.route('/fileManager', methods=['GET', 'POST'])
+@authManager.requireAuthentication
+def fileManager():
+    user = authManager.getUserData()
+    if request.method == 'POST':
+        # do stuff when the form is submitted
+        if not user:
+            abort(403)
+        id = user['id']
+
+        if 'file' not in request.files:
+            flash('No file part')
+            return redirect(request.url)
+        file = request.files['file']
+        # if user does not select file, browser also
+        # submit a empty part without filename
+        if file.filename.strip() == '':
+            flash('No selected file')
+            return redirect(request.url)
+        if file:
+            # Determine shortened file name (secure)
+            filename = secure_filename(file.filename.strip())
+            while (len(filename) > 50):
+                cutString = len(filename)%50
+                filename = filename[cutString:len(filename)]
```

Comment:
Why do filenames need to be limited to 50 chars?

Reply:
This is the data.error I encounter when I set the limit is 60. 50 works fine so I will keep it